engine/lib/tags.php

```diff
@@ -13,9 +13,9 @@
  *
  * @param string $string Comma-separated tag string
  *
- * @return array|false An array of strings, or false on failure
+ * @return array|false|mixed An array of strings, false if no data provided, or the original data if input was not a string
  */
-function string_to_tag_array($string) {
+function string_to_tag_array($string = false) {
```

Comment:
In fact I think I'm wrong to suggest changing the var name. It should be $string, but we should handle the invalid input first:
```php
if (!is_string($string)) {
    return $string;
}
// do the split...
```